src/Mod/Sketcher/App/planegcs/Constraints.h

```diff
@@ -167,10 +167,38 @@ class SketcherExport Constraint
         return internalAlignment;
     }
 
+
     virtual ConstraintType getTypeId();
     virtual void rescale(double coef = 1.);
-    virtual double error();
-    virtual double grad(double*);
+
+    // error and gradient combined. Values are returned through pointers.
+    virtual void errorgrad(double* err, double* grad, double* param)
+    {
+        (void)param;
+        if (err) {
+            *err = 0.;
+        }
+        if (grad) {
+            *grad = 0.;
+        }
+    };
+    virtual double error()
```

Comment:
Does this still need to be a virtual now? Are there still overrides?

Reply:
Yes some constraints doesn't use the `errorgrad` function but separate `error` and `grad`